### PR TITLE
HDS-1484-number-input-scroll

### PR DESCRIPTION
**Pull request by mrTuomoK**
## Description
Adds throttling to mouse-scroll event in numberInput

## Related Issue
[HDS-1484](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1484)

## Motivation and Context
Prevents changing the number input value too quickly by scrolling on the field (either purposely or not)

## How Has This Been Tested?
- Local machine with mouse wheel & touchPad
- On Safari, Chrome & Firefox (Firefox doesn't even scroll in number inputs so no difference in this case)

## Preview video
This is quite accurate gif how fast the value changes no matter how fast the user now scrolls on the input field.
scroll

[HDS-1484]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1484?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/react/src/components/numberInput/NumberInput.tsx [hunk 1]
```diff
@@ -117,6 +118,29 @@ export const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>(
     const notifyScreenReaderStepperChangedValue = () => {
       setScreenReaderValue(String(inputRef.current.value));
     };
+    let throttledMouseWheel = false;
```
Comment (mikkojamG) [inline packages/react/src/components/numberInput/NumberInput.tsx:121]: Noice 👍 Could this be written as a custom hook for cleaner syntax? 🤔  E.g. `useThrottledMouseWeel(inputRef)`

Comment (mrTuomoK) [reply packages/react/src/components/numberInput/NumberInput.tsx:121]: Will check & test 👍

File: packages/react/src/hooks/useThrottledWheel.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+/*
+  A Hook that will throttle the wheel event
+  for 200ms on given html input reference object.
+*/
+function useThrottledWheel(inputRef: React.MutableRefObject<HTMLInputElement>) {
+  let throttledWheel = false;
+
+  const throttledWheelToggler = throttle(() => {
+    throttledWheel = false;
+  }, 200);
```
[the remark this answers is not in the document]
Comment (mrTuomoK) [reply packages/react/src/hooks/useThrottledWheel.ts:12]: Would someone like it? 🤷 or? ... well I can make it so... and use the 200ms as default the 👍

[the remark this answers is not in the document]
Comment (mikkojamG) [reply packages/react/src/hooks/useThrottledWheel.ts:12]: Maybe make it optional and keep 200ms as a default? 🙂

**Review by mikkojamG — APPROVED** [review]
Beautiful 💯

**Review by harriplappalainen — APPROVED** [review]
Great work! 🚀

**Pull request merged**